packages/providers/src/utils/receipts.ts

```diff
@@ -35,3 +59,212 @@ export const getReceiptsWithMissingData = (receipts: Array<TransactionResultRece
       missingOutputContractIds: [],
     }
   );
+
+const hexOrZero = (hex?: string | null) => hex || ZeroBytes32;
+
+export function assembleReceiptByType(receipt: GqlReceiptFragmentFragment) {
+  switch (receipt.receiptType) {
+    case GqlReceiptType.Call: {
+      const callReceipt: ReceiptCall = {
+        type: ReceiptType.Call,
+        from: hexOrZero(receipt.contract?.id),
+        to: hexOrZero(receipt?.to?.id),
+        amount: bn(receipt.amount),
+        assetId: hexOrZero(receipt.assetId),
+        gas: bn(receipt.gas),
+        param1: bn(receipt.param1),
+        param2: bn(receipt.param2),
+        pc: bn(receipt.pc),
+        is: bn(receipt.is),
+      };
+
+      return callReceipt;
+    }
+
+    case GqlReceiptType.Return: {
+      const returnReceipt: ReceiptReturn = {
+        type: ReceiptType.Return,
+        id: hexOrZero(receipt.contract?.id),
+        val: bn(receipt.val),
+        pc: bn(receipt.pc),
+        is: bn(receipt.is),
+      };
+
+      return returnReceipt;
+    }
+
+    case GqlReceiptType.ReturnData: {
+      const returnDataReceipt: ReceiptReturnData = {
+        type: ReceiptType.ReturnData,
+        id: hexOrZero(receipt.contract?.id),
+        ptr: bn(receipt.ptr),
+        len: bn(receipt.len),
+        digest: hexOrZero(receipt.digest),
+        pc: bn(receipt.pc),
+        is: bn(receipt.is),
+      };
+
+      return returnDataReceipt;
+    }
+
+    case GqlReceiptType.Panic: {
+      const panicReceipt: ReceiptPanic = {
+        type: ReceiptType.Panic,
+        id: hexOrZero(receipt.contract?.id),
+        reason: bn(receipt.reason),
+        pc: bn(receipt.pc),
+        is: bn(receipt.is),
+        contractId: hexOrZero(receipt.contractId),
+      };
```

Comment:
The issue https://github.com/FuelLabs/fuels-ts/issues/956 was happening because of the absence of `contractId` in the Panic Receipt `rawpayload` property, which was causing `estimateTxDependencies` failing to identify the `contractId` and add it to the transaction request.

However, with the removal of the `rawpayload` property in `fuel-core 0.20`, the way we assemble receipts has changed. We no longer decode them using the `rawpayload`, but instead assemble them using the Receipt's existing properties from the GraphQL response. 

As a result of the `contractId` property being present on the GraphQL response, `estimateTxDependencies` now successfully identifies and adds missing contract ids to the transaction request.

This change resolves the issue, so I am closing https://github.com/FuelLabs/fuels-ts/issues/956 with this PR.